tests/samples/matmul_peeled_objectfifo.mlir

```diff
@@ -0,0 +1,116 @@
+// RUN: iree-opt --pass-pipeline="builtin.module(fold-memref-alias-ops,iree-amdaie-pack-to-dma,air-copy-to-dma,iree-amdaie-air-dma-to-amdaie-dma,iree-amdaie-insert-aie-workgroup,iree-amdaie-fuse-logicalobjectfifo-into-workgroup,cse,iree-amdaie-unroll-and-distribute-workgroup,cse,iree-amdaie-dma-to-circular-dma,func.func(iree-amdaie-create-aie-workgroup),cse,iree-amdaie-canonicalize-doubly-strided-op,iree-amdaie-consume-produce-to-acquire-release,cse,canonicalize,iree-amdaie-controlcode-loop-unroll,cse,canonicalize,canonicalize,iree-amdaie-lower-to-aie,canonicalize)" --split-input-file %s | FileCheck %s
+
+// CHECK:       aie.device(npu1)
+// CHECK-DAG:   %[[TILE_0_2:.+]] = aie.tile(0, 2)
+// CHECK-DAG:   %[[TILE_1_2:.+]] = aie.tile(1, 2)
+// CHECK-DAG:   aie.objectfifo @[[OBJ0:.+]](%[[TILE_0_2]]
+// CHECK-DAG:   aie.objectfifo @[[OBJ1:.+]](%[[TILE_1_2]]
+// CHECK-DAG:   aie.core(%[[TILE_0_2]])
+// CHECK:         aie.objectfifo.acquire @[[OBJ0]](Produce, 1)
+// CHECK-DAG:   aie.core(%[[TILE_1_2]])
+// CHECK:         aie.objectfifo.acquire @[[OBJ1]](Produce, 1)
+// CHECK:       func.func @sequence(%[[ARG0:.+]]: memref<32x1024xi32>, %[[ARG1:.+]]: memref<1024x64xi32>, %[[ARG2:.+]]: memref<32x64xi32>)
+// CHECK-DAG:     aiex.npu.dma_memcpy_nd
+// CHECK-SAME:    %[[ARG0]]
+// CHECK-DAG:     aiex.npu.dma_memcpy_nd
+// CHECK-SAME:    %[[ARG1]]
+// CHECK-DAG:     aiex.npu.dma_memcpy_nd(0, 0, %[[ARG2]][1, 1, 0, 0][1, 1, 32, 32][1, 1, 64]
+// CHECK-DAG:     aiex.npu.dma_memcpy_nd(0, 0, %[[ARG2]][1, 1, 0, 32][1, 1, 32, 32][1, 1, 64]
+#map = affine_map<(d0) -> (d0 * 32)>
+#map1 = affine_map<(d0) -> (d0 * 64)>
+#map2 = affine_map<(d0, d1, d2, d3, d4, d5) -> (d2, d0, d3, d5)>
+#map3 = affine_map<(d0, d1, d2, d3, d4, d5) -> (d1, d2, d5, d4)>
+#map4 = affine_map<(d0, d1, d2, d3, d4, d5) -> (d1, d0, d3, d4)>
+module {
+  func.func @matmul_i32() {
+    %c64 = arith.constant 64 : index
+    %c960 = arith.constant 960 : index
+    %c0_i32 = arith.constant 0 : i32
+    %c0 = arith.constant 0 : index
+    %0 = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) alignment(64) offset(%c0) flags(ReadOnly) : memref<1024x64xi32>
+    memref.assume_alignment %0, 64 : memref<1024x64xi32>
+    %1 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(64) offset(%c0) flags(ReadOnly) : memref<32x1024xi32>
+    memref.assume_alignment %1, 64 : memref<32x1024xi32>
+    %2 = hal.interface.binding.subspan set(0) binding(2) type(storage_buffer) alignment(64) offset(%c0) : memref<32x64xi32>
+    memref.assume_alignment %2, 64 : memref<32x64xi32>
+    %alloc = memref.alloc() : memref<4x8x8x8xi32, 2>
+    %alloc_0 = memref.alloc() : memref<8x8x4x8xi32, 2>
+    %alloc_1 = memref.alloc() : memref<4x8x4x8xi32, 2>
+    %alloc_2 = memref.alloc() : memref<32x32xi32, 1>
+    %alloc_3 = memref.alloc() : memref<64x32xi32, 1>
+    %alloc_4 = memref.alloc() : memref<32x64xi32, 1>
```

Comment:
These require first level to be pad instead of pack. Does objectfifo potentially work with more dimensions?

Reply:
Yeah, that should be fine. I created this example when the main pipeline was `pad-pack`.